### PR TITLE
Mangata dev v0.9.28

### PR DESCRIPTION
**Pull request by devdanco**

### CODE REVIEW

File: frame/support/procedural/src/construct_runtime/expand/call.rs [hunk 1]
```diff
@@ -163,19 +163,6 @@ pub fn expand_outer_dispatch(
...
-				#scrate::storage::with_storage_layer(|| {
-					#scrate::dispatch::Dispatchable::dispatch(self, origin)
-				})
-			}
-			fn dispatch_bypass_filter_with_storage_layer(self, origin: Origin) -> #scrate::dispatch::DispatchResultWithPostInfo {
-				#scrate::storage::with_storage_layer(|| {
-					#scrate::traits::UnfilteredDispatchable::dispatch_bypass_filter(self, origin)
-				})
-			}
-		}
```
Comment (iStrike7) [inline frame/support/procedural/src/construct_runtime/expand/call.rs:178]: This needn't have been removed

Comment (iStrike7) [reply frame/support/procedural/src/construct_runtime/expand/call.rs:178]: Was it removed in origin substrate as well?

File: frame/support/src/traits/dispatch.rs [hunk 2]
```diff
@@ -236,23 +236,6 @@ pub trait UnfilteredDispatchable {
...
-	fn dispatch_with_storage_layer(self, origin: Self::Origin) -> DispatchResultWithPostInfo;
-
-	/// Same as `dispatch_bypass_filter` from the [`UnfilteredDispatchable`] trait, but specifically
-	/// spawns a new storage layer to execute the call inside of.
-	fn dispatch_bypass_filter_with_storage_layer(
-		self,
-		origin: Self::Origin,
-	) -> DispatchResultWithPostInfo;
-}
-
```
Comment (iStrike7) [inline frame/support/src/traits/dispatch.rs:255]: This needn't have been removed.
Was it removed in the substrate origin implementation?

File: frame/support/src/dispatch.rs [hunk 3]
```diff
@@ -1473,9 +1472,9 @@ macro_rules! decl_module {
...
 		$mod_type:ident<$trait_instance:ident $(, $instance:ident)?> $fn_name:ident $origin:ident $system:ident [ $( $param_name:ident),* ]
 	) => {
-			// We execute all dispatchable in at least one storage layer, allowing them
+			// We execute all dispatchable in a new storage layer, allowing them
 			// to return an error at any point, and undoing any storage changes.
-			$crate::storage::in_storage_layer(|| {
+			$crate::storage::with_storage_layer(|| {
 				<$mod_type<$trait_instance $(, $instance)?>>::$fn_name( $origin $(, $param_name )* ).map(Into::into).map_err(Into::into)
 			})
 	};
```
Comment (iStrike7) [inline frame/support/src/dispatch.rs:1480]: This should not be wrapped in storage layer. This shouldn't have been there in .26 either. We had missed it. Please remove it.

File: frame/support/src/dispatch.rs [hunk 4]
```diff
@@ -1788,9 +1787,11 @@ macro_rules! decl_module {
...
 			$origin: $origin_ty $(, $param: $param_ty )*
 		) -> $crate::dispatch::DispatchResult {
-			$crate::sp_tracing::enter_span!($crate::sp_tracing::trace_span!(stringify!($name)));
-			{ $( $impl )* }
-			Ok(())
+			$crate::storage::with_storage_layer(|| {
+				$crate::sp_tracing::enter_span!($crate::sp_tracing::trace_span!(stringify!($name)));
+				{ $( $impl )* }
+				Ok(())
+			})
```
Comment (iStrike7) [inline frame/support/src/dispatch.rs:1794]: This should not be wrapped in storage layer. This shouldn't have been there in .26 either. We had missed it. Please remove it.

File: frame/support/src/dispatch.rs [hunk 5]
```diff
@@ -1806,8 +1807,10 @@ macro_rules! decl_module {
 	) => {
 		$(#[$fn_attr])*
 		$vis fn $name($origin: $origin_ty $(, $param: $param_ty )* ) -> $result {
-			$crate::sp_tracing::enter_span!($crate::sp_tracing::trace_span!(stringify!($name)));
-			$( $impl )*
+			$crate::storage::with_storage_layer(|| {
+				$crate::sp_tracing::enter_span!($crate::sp_tracing::trace_span!(stringify!($name)));
+				$( $impl )*
+			})
```
Comment (iStrike7) [inline frame/support/src/dispatch.rs:1813]: This should not be wrapped in storage layer. This shouldn't have been there in .26 either. We had missed it. Please remove it.

**Review by iStrike7 — CHANGES_REQUESTED** [review]

File: frame/support/procedural/src/construct_runtime/expand/call.rs (hunk 1 above)
Comment (devdanco) [reply frame/support/procedural/src/construct_runtime/expand/call.rs:178]: yes it was removed

**Review by iStrike7 — APPROVED** [review]

**Pull request merged**